internal/codespaces/api/api.go

```diff
@@ -188,36 +189,54 @@ type CodespaceEnvironmentConnection struct {
 
 // ListCodespaces returns a list of codespaces for the user.
 func (a *API) ListCodespaces(ctx context.Context) ([]*Codespace, error) {
-	req, err := http.NewRequest(
-		http.MethodGet, a.githubAPI+"/user/codespaces", nil,
-	)
-	if err != nil {
-		return nil, fmt.Errorf("error creating request: %w", err)
-	}
+	page := 1
+	per_page := 50
+	codespaces := []*Codespace{}
 
-	a.setHeaders(req)
-	resp, err := a.do(ctx, req, "/user/codespaces")
-	if err != nil {
-		return nil, fmt.Errorf("error making request: %w", err)
-	}
-	defer resp.Body.Close()
+	for {
```

Comment:
```suggestion
	for page := 1;; page++ {
```

And you can remove the declaration above